packages/core/core/src/RequestTracker.js

```diff
@@ -84,45 +88,61 @@ type SerializedRequestGraph = {|
 |};
 
 type FileNode = {|id: ContentKey, +type: 'file'|};
+
 type GlobNode = {|id: ContentKey, +type: 'glob', value: InternalGlob|};
+
 type FileNameNode = {|
   id: ContentKey,
   +type: 'file_name',
-  value: string,
 |};
+
 type EnvNode = {|
   id: ContentKey,
   +type: 'env',
-  value: {|key: string, value: string | void|},
+  value: string | void,
 |};
 
 type OptionNode = {|
   id: ContentKey,
   +type: 'option',
-  value: {|key: string, hash: string|},
+  hash: string,
 |};
 
 type Request<TInput, TResult> = {|
   id: string,
-  +type: string,
+  +type: RequestType,
   input: TInput,
   run: ({|input: TInput, ...StaticRunOpts<TResult>|}) => Async<TResult>,
 |};
 
-type StoredRequest = {|
-  id: string,
-  +type: string,
-  result?: mixed,
-  resultCacheKey?: ?string,
-|};
-
 type InvalidateReason = number;
 type RequestNode = {|
   id: ContentKey,
   +type: 'request',
-  value: StoredRequest,
+  +requestType?: RequestType,
   invalidateReason: InvalidateReason,
+  result?: mixed,
+  resultCacheKey?: ?string,
+  hash?: string,
 |};
+
+export type RequestType =
+  | 'parcel_build_request'
+  | 'bundle_graph_request'
+  | 'asset_graph_request'
+  | 'entry_request'
+  | 'target_request'
+  | 'parcel_config_request'
+  | 'path_request'
+  | 'dev_dep_request'
+  | 'asset_request'
+  | 'config_request'
+  | 'write_bundles_request'
+  | 'package_request'
+  | 'write_bundle_request'
+  | 'validation_request'
+  | string;
```

Comment:
I'm iffy about this `string` here. I added it to appease Flow because we use a `mock_request` type in the RequestTracker's unit tests, but I wonder if I should just flow-ignore those instead.

Reply:
I agree, let's flow ignore the test rather than the main code.